source/common/config/filter_json.cc

```diff
@@ -399,5 +399,58 @@ void FilterJson::translateClientSslAuthFilter(
                                       *proto_config.mutable_ip_white_list());
 }
 
+void FilterJson::translateGzipFilter(const Json::Object& json_config,
+                                     envoy::api::v2::filter::http::Gzip& proto_config) {
+  json_config.validateSchema(Json::Schema::GZIP_HTTP_FILTER_SCHEMA);
+
+  proto_config.set_disable_on_etag_header(json_config.getBoolean("disable_on_etag_header", false));
+  proto_config.set_disable_on_last_modified_header(
+      json_config.getBoolean("disable_on_last_modified_header", false));
+
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, content_length);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, memory_level);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, window_bits);
+
+  for (const auto& json_content_type : json_config.getStringArray("content_type", true)) {
+    auto* content_type = proto_config.mutable_content_type()->Add();
+    *content_type = json_content_type;
+  }
+
+  static const struct {
+    const char* level_str;
+    envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum level_enum;
+  } levels_map[] = {
+      {"default", envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum_DEFAULT},
+      {"best", envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum_BEST},
+      {"speed", envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum_SPEED},
+  };
+
+  const auto json_compression_level = json_config.getString("compression_level", "default");
+  for (const auto& level_map : levels_map) {
+    if (level_map.level_str == json_compression_level) {
+      proto_config.set_compression_level(level_map.level_enum);
+      break;
+    }
+  }
+
+  static const struct {
+    const char* strategy_str;
+    envoy::api::v2::filter::http::Gzip_CompressionStrategy strategy_enum;
+  } strategies_map[] = {
+      {"default", envoy::api::v2::filter::http::Gzip_CompressionStrategy_DEFAULT},
+      {"filtered", envoy::api::v2::filter::http::Gzip_CompressionStrategy_FILTERED},
+      {"huffman", envoy::api::v2::filter::http::Gzip_CompressionStrategy_HUFFMAN},
+      {"rle", envoy::api::v2::filter::http::Gzip_CompressionStrategy_RLE},
+  };
```

Comment:
thanks; looks great!  One other optional tweak is that you could templatize this enum struct for brevity given there are two similar ones:

```
template<typename Enum> struct NameAndEnum {
  const char* name;
  Enum enum;
};

static const NameAndEnum<envoy::api::v2::filter::http::Gzip_envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum> levels_map[] = { ...};
static const NameAndEnum<envoy::api::v2::filter::http::Gzip_CompressionStrategy> strategies_map[] = { ...};
```